cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -300,7 +299,13 @@ private void upgradeCoProcessor(TableId tableId, Class<? extends Coprocessor> co
       }
 
       // Add new coprocessor
-      addCoprocessor(coprocessor, newDescriptor);
+      CoprocessorDescriptor coprocessorDescriptor =
+        coprocessorManager.getCoprocessorDescriptor(coprocessor, Coprocessor.PRIORITY_USER);
+      Path path = coprocessorDescriptor.getPath() == null ? null : new Path(coprocessorDescriptor.getPath());
+      tableDescriptor.addCoprocessor(coprocessorDescriptor.getClassName(),
+                                     path,
```

Comment:
Not introduced in this PR but, do we need `manageCoprocessor` flag in this class? seems unused.